sample-xml/app/src/main/java/co/nimblehq/sample/xml/ui/screens/second/SecondViewModel.kt

```diff
@@ -0,0 +1,38 @@
+package co.nimblehq.sample.xml.ui.screens.second
+
+import co.nimblehq.sample.xml.domain.usecase.UseCase
+import co.nimblehq.sample.xml.domain.usecase.UseCaseResult
+import co.nimblehq.sample.xml.model.UiModel
+import co.nimblehq.sample.xml.model.toUiModels
+import co.nimblehq.sample.xml.ui.base.BaseViewModel
+import co.nimblehq.sample.xml.util.DispatchersProvider
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import javax.inject.Inject
+
+@HiltViewModel
+class SecondViewModel @Inject constructor(
+    private val useCase: UseCase,
+    dispatchers: DispatchersProvider
+) : BaseViewModel(dispatchers) {
+
+    private val _uiModels = MutableStateFlow<List<UiModel>>(emptyList())
+    val uiModels: StateFlow<List<UiModel>>
+        get() = _uiModels
+
+    init {
+        execute {
+            when (val result = useCase.execute()) {
```

Comment:
I don't think we need to call `useCase` on the second screen according to our implementation plan 🤔